File: LeetCode/python/ValidAnagram.py
class Solution:
    # @param {string} s
    # @param {string} t
    # @return {boolean}
    def isAnagram(self, s, t):
        if(len(s) != len(t)):
            return False
		
		#根据字符串s构建字典	
        wordDict = {}
        for c in s:
            if c in wordDict:
                wordDict[c] += 1
            else:
                wordDict[c] = 1
		
		#查找s中的单词是否都出现在t中了
        for c in t:
            if c in wordDict:
                if(wordDict[c] == 0):
                    return False
			
                wordDict[c] -= 1
            else:
                return False
		
        return True				
			
#测试函数
s = Solution()
print(s.isAnagram("f",""))		

File: LeetCode/iter2/C++/ValidAnagram.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <vector>
#include <sstream>
#include <unordered_map>
#include <unordered_set>
#include <utility>
#include <queue>
#include <stack>

using namespace std;

class Solution {
public:
	bool isAnagram(string s, string t) {
		if (s.size() != t.size())
			return false;

		vector<int> alphabat(26,0);

		for (char ch : s)
			alphabat[ch - 'a']++;

		bool ret = true;
		for (char ch : t)
		{
			if(alphabat[ch-'a']==0)
			{
				ret = false;
				break;
			}

			alphabat[ch - 'a']--;
		}

		return ret;
	}
};

void testPrint(string s, string t)
{
	Solution solution;
	if (solution.isAnagram(s, t))
		cout << s << " " << t << " is anagram\n";
	else
		cout << s << " " << t << " is not anagram\n";
}

int main(void)
{

	testPrint("rat","car");
	testPrint("anagram","nagaram");


	return 0;
}